language/move-prover/docgen/src/docgen.rs

```diff
@@ -664,7 +691,179 @@ impl<'env> Docgen<'env> {
+    /// Generate a static call diagram (.svg) starting from the given function.
+    fn gen_call_diagram(&self, fun_id: QualifiedId<FunId>) {
+        let fun_env = self.env.get_function(fun_id);
+        let name_of = |env: &FunctionEnv| {
+            if fun_env.module_env.get_id() == env.module_env.get_id() {
+                env.get_simple_name_string()
+            } else {
+                Rc::from(format!("\"{}\"", env.get_name_string()))
+            }
+        };
+
+        let mut dot_src: String = String::from("digraph G {\n");
+        let mut visited: BTreeSet<QualifiedId<FunId>> = BTreeSet::new();
+        let mut queue: VecDeque<QualifiedId<FunId>> = VecDeque::new();
+
+        visited.insert(fun_id);
+        queue.push_back(fun_id);
+
+        while let Some(id) = queue.pop_front() {
+            let caller_env = self.env.get_function(id);
+            let caller_name = name_of(&caller_env);
+            let callee_list = caller_env.get_called_functions();
+
+            if fun_env.module_env.get_id() == caller_env.module_env.get_id() {
+                dot_src.push_str(&format!("    {}\n", caller_name));
+            } else {
+                let module_name = caller_env
+                    .module_env
+                    .get_name()
+                    .display(caller_env.module_env.symbol_pool());
+                dot_src.push_str(&format!("    subgraph cluster_{} {{\n", module_name));
+                dot_src.push_str(&format!("        label = \"{}\";\n", module_name));
+                dot_src.push_str(&format!(
+                    "        {}[label=\"{}\"]\n",
+                    caller_name,
+                    caller_env.get_simple_name_string()
+                ));
+                dot_src.push_str("    }\n");
+            }
+
+            for callee_id in callee_list.iter() {
+                let callee_env = self.env.get_function(*callee_id);
+                let callee_name = name_of(&callee_env);
+                dot_src.push_str(&format!("    {} -> {}\n", caller_name, callee_name));
+                if !visited.contains(callee_id) {
+                    visited.insert(*callee_id);
+                    queue.push_back(*callee_id);
+                }
+            }
+        }
+        dot_src.push_str("}\n");
+
+        let out_file_path = PathBuf::from(&self.options.output_directory)
+            .join("img")
+            .join(format!(
+                "{}_call_graph.svg",
+                fun_env.get_name_string().to_string().replace("::", "_")
+            ));
+
+        self.gen_svg_file(&out_file_path, &dot_src);
+    }
+
+    /// Generate a forward (or backward) dependency diagram (.svg) for the given module.
+    fn gen_dependency_diagram(&self, module_id: ModuleId, is_forward: bool) {
+        let module_env = self.env.get_module(module_id);
+        let module_name = module_env.get_name().display(module_env.symbol_pool());
+
+        let mut dot_src: String = String::from("digraph G {\n");
```

Comment:
Same question/possible update to `dot_src` here as in `gen_call_diagram`

Reply:
Done.